user/friends_management.py

```diff
@@ -94,6 +94,7 @@ def remove_friend(self, req_item: RemoveFriendRequest):
         if self.validate_username(friends_username):
             try:
                 self.db.remove_friend(username, friends_username)
+                req_item.set_response()
```

Comment:
This was the reason the test failed (in addition to not having delete privileges earlier). Previously the response was not set when the operation was successful. I must have missed it when I was reworking the message item stuff.